web/scripts/widgets.js

```diff
@@ -10,37 +10,167 @@ function getNumberDefaults(inputData, defaultStep) {
 	return { val: defaultVal, config: { min, max, step: 10.0 * step } };
 }
 
-export function addRandomizeWidget(node, targetWidget, name, defaultValue = false) {
-	const randomize = node.addWidget("toggle", name, defaultValue, function (v) {}, {
-		on: "enabled",
-		off: "disabled",
+export function addValueControlWidget(node, targetWidget, defaultValue = "randomize", values) {
+	const valueControl = node.addWidget("combo", "control_after_generate", defaultValue, function (v) { }, {
+		values: ["fixed", "increment", "decrement", "randomize"],
 		serialize: false, // Don't include this in prompt.
-	});
+	})
+	valueControl.afterQueued = () => {
+
+		var v = valueControl.value;
+		var w = targetWidget.name;
+
+		let min = 0.0;
+		let max = targetWidget.options?.max;
+		let range = Math.max(min, max);
+
+
+		//set the max/min values depending on the parameter
+		switch (w) {
+			case ("seed"):
+			case ("noise_seed"):
+				console.log("noise_seed/seed");
+				max = 4294967294; // limit max to something that javascript can handle
+				range = Math.max(min, max);
+				break;
+			case ("cfg"):
+				console.log("cfg");
+				max = 50.0;
+				range = Math.max(min, max);
+				break;
+			case ("steps"):
+				max = 100;
+				min = 1;
+				range = Math.max(min, max);
+				break;
+			case ("start_at_step"):
+				max = 50000;
+				range = Math.max(min, max);
+				break;
+			case ("end_at_step"):
+				min = 50000;
+				max = 100000;
+				range = Math.max(min, max);
+				break;
+			case ("denoise"):
+				console.log("denoise");
+				max = 1.0;
+				min = 0.001;
+				range = Math.max(min, max);
+				break;
+			case ("height"):
+			case ("width"):
+				console.log("width/height");
+				min = 64;
+				max = 1920;
+				range = Math.max(0, Math.floor((max - min) / 64));
+				break;
+			default:
+				console.log("default (Failed)");
+				break;
+		}
 
-	randomize.afterQueued = () => {
-		if (randomize.value) {
-			const min = targetWidget.options?.min;
-			let max = targetWidget.options?.max;
```

Comment:
Why did you hardcode the values instead of just using this?

Reply:
the default values were causing unexpected behaviour for multiple values. Other set max values were simple too high (for example steps beyond 1000) and would cause render times that would be simply too long for anyone to reasonably complete. The other reason for this is I plan on replacing these values with user inputs for custom ranges. The settings I picked here complete generations within a reasonable time-frame and have enough variation for seeing the effects of different parameters. If you would like, I can remove them or just add the user inputs.

Reply:
it's better from a code quality perspective to not hardcode anything and use the values from the backend because it's easier to maintain.

Reply:
In case you don't already know, you can access a widget's native step size like this:
```const step_size = targetWidget.options?.step / 10;```

Reply:
I've removed setting the min/max values as requested.
>const step_size = targetWidget.options?.step / 10;
I could add this if you want comfy, that way something unreasonable like 4k steps. Also, consider changing the max seed as brought up in #352. Other than that it's good to go.